### ISSUE TITLE
Better way to structure the repo?

### ISSUE THREAD

**Issue by bakerwho** [aimacode/aima-python#772]
Perhaps it's just me, but it looks pretty hard to navigate through the code while reading through the book. I have to open and read [agents.py](https://github.com/bakerwho/aima-python/blob/master/agents.py) to see that it discusses code from Chapters 1 and 2.

There's a notebook called [intro.pynb](https://github.com/bakerwho/aima-python/blob/master/intro.ipynb)  that gives some information about the reasons for this - 
> A typical module in the repository has the code for a single chapter in the book, but some modules combine several chapters.
But this notebook is not referenced in the README. Besides, it may be nice to have a companion to the book that links, chapter by chapter, section by section, what files or notebooks might be relevant.

Similarly, the [instructions](https://github.com/aimacode/aima-python/blob/master/SUBMODULE.md) to deal with the [aima-data](https://github.com/aimacode/aima-data/tree/c81e8907917c60bfaedccc720c6b8ce07fabb222) submodule - are not mentioned anywhere in the README. The emptiness of the aima-data subfolder after a `git clone` might be confusing for newcomers to git. If they move files into the folder from local downloads, it might also complicate the merging of their changes (not 100% sure about this, but it looks like it might be a problem).

I'd like to open a discussion on this before I make any changes, but what do you think, @norvig, @MrDupin?

**Comment by MrDupin** [comment]
You are right that we do not have a clear installation guide. I have amended that in #773, thanks for pointing it out!

The instructions in SUBMODULE are simply for updating the submodule to the latest version, so that users can download the correct datasets. It is not something that users need to do themselves, it is more of a reminder for admins.

About the structure, I think it's fine as it is. We already have the README that points people to the right direction. It is up to Mr. Norvig, but I personally wouldn't change this.

**Comment by JaeDukSeo** [comment]
I tried to clean it up, but it wasn't the right approach. So I guess we'll have to wait.

**Comment by bakerwho** [comment]
Might it make sense to put the notebooks in a folder called 'notebooks' and the `.py` files in a folder called 'bin' or 'pycode' or 'scripts'? The repo is still a bit messy. Scrolling through it, one finds files like `pytest.ini` and `requirements.txt` in midst of `ipynb` files that explain concepts! Is there a reason such restructuring has not already been done?

If we're not taking any further steps in this direction, I reckon this issue can be closed.

**Comment by dmeoli** [comment]
Repo-structure discussion from 2018; no concrete actionable change converged. Closing as stale — a specific proposal PR would be the way to revisit.
